### PR TITLE
[BEAM-7173] Avoiding schema autodetection by default in WriteToBigQuery

### PR DESCRIPTION
**Pull request by pabloem**
@tvalentyn 
cc: @aaltay

### CODE REVIEW

**Comment by pabloem** [comment]
Run Python PreCommit

**Comment by pabloem** [comment]
Run Python PostCommit

**Comment by tvalentyn** [comment]
Run Python Postcommit

File: sdks/python/apache_beam/io/gcp/bigquery.py [hunk 1]
```diff
@@ -666,7 +666,12 @@ def _create_table_if_needed(self, table_reference, schema=None):
     logging.debug('Creating or getting table %s with schema %s.',
                   table_reference, schema)
 
-    table_schema = self.get_table_schema(schema)
+    if schema == SCHEMA_AUTODETECT:
```
Comment (tvalentyn) [inline sdks/python/apache_beam/io/gcp/bigquery.py]: Are we assuming that this helper only gets called only in streaming inserts codepath? 
Perhaps we should strengthen the condition to something like: `if schema == SCHEMA_AUTODETECT and  method == STREAMING_INSERTS`, or move this check somewhere up the call stack to fail early?

File: sdks/python/apache_beam/io/gcp/bigquery_test.py [hunk 2]
```diff
@@ -531,6 +531,7 @@ def test_value_provider_transform(self):
            | "WriteWithMultipleDests2" >> beam.io.gcp.bigquery.WriteToBigQuery(
                table=value_provider.StaticValueProvider(
                    str, '%s:%s' % (self.project, output_table_2)),
+               schema=bigquery.SCHEMA_AUTODETECT,
```
Comment (tvalentyn) [inline sdks/python/apache_beam/io/gcp/bigquery_test.py]: Note that SCHEMA_AUTODETECT if BQ has to ingest `bytes` data into a table that has `bytes` field.
Also, should we have a test with and without autodetect behavior?

**Comment by tvalentyn** [comment]
I think we should loop in Cham, who reviewed previous BQ PRs, perhaps once the tests pass.
//cc: @chamikaramj

**Comment by pabloem** [comment]
Run Python PostCommit

**Comment by pabloem** [comment]
Run Python PostCommit

**Comment by pabloem** [comment]
Run Python PostCommit

**Comment by pabloem** [comment]
Run Python PostCommit

**Comment by pabloem** [comment]
Run Portable_Python PreCommit

**Comment by pabloem** [comment]
Run Python Precommit

**Comment by pabloem** [comment]
Run Python PostCommit

**Comment by pabloem** [comment]
Run Python PostCommit

**Comment by pabloem** [comment]
Run Python PostCommit

**Comment by pabloem** [comment]
Run Python PostCommit

File: sdks/python/apache_beam/io/gcp/bigquery_test.py (hunk 2 above)
Comment (pabloem) [reply sdks/python/apache_beam/io/gcp/bigquery_test.py]: There are tests with and without this feature running in that file. WDYT?

Comment (tvalentyn) [reply sdks/python/apache_beam/io/gcp/bigquery_test.py]: I think that should be fine and we don't need to duplicate every test twice, but you might want to do a one-off run of integration tests where the value of autodetection flag is changed to the opposite value, to make sure there are no surprises.

**Comment by tvalentyn** [comment]
LGTM, thank you! Please squash commits before the merge.

**Pull request merged**